sdk/storage/Azure.Storage.Blobs/src/Models/BlockList.cs

```diff
@@ -57,6 +57,12 @@ internal static partial class BlobExtensions
         /// <returns>The BlockList response.</returns>
         internal static Response<BlockList> ToBlockList(this Response<GetBlockListOperation> response)
         {
+            // Return an exploding Response on 304
+            if (response.IsExplodingResponse())
+            {
+                return response.GetRawResponse().AsExplodingResponse<BlockList>();
```

Comment:
That's an affectionate name for `Response<T>`s that will throw if you try to access `Value`.  It's the pattern we've settled on for `304`s per https://github.com/Azure/azure-sdk/issues/638.

We want to avoid throwing exceptions if we can because they're not great for performance and clutter up logs.  If you add an HTTP precondition on a `GET`/`HEAD` operation, Storage will return a `304`.  Instead of throwing, we'll give you back a `Response<T>` that has a `Status == 304` and will still throw if you try to access `Response<T>.Value`.  This gives us the best of both worlds because it'll still throw for customers who don't check, but there's a way for customers who know better to avoid the exception.  All other HTTP verbs will return a `Status == 412` and we always throw for those.

We just need to be careful in our code that we don't accidentally touch `Response<T>.Value` if it might explode.